uji.py

```diff
@@ -1054,6 +1065,152 @@ def upload(self):
+
+    def _sanitize(self, line):
+        return line.replace('\t', '    ')
+
+    def execute_command(self):
+        line = self.lines[self.cursor_offset]
+        if not self.is_checkbox(line) or "⚙" not in line:
+            return
+
+        # list of regex to deduce the type of command to be run
+        command_re = {
+                'attach': r'^(\s*)- \[.\].* \[`(.*)`\]\((.*)\).*',
+                'human': r'^(\s*)- \[.\].* `(.*)`: <strong>(.*)</strong>$',
+                'single': r'^(\s*)- \[.\].* `(.*)`: `(.*)`$',
+                'multi': r'^(\s*)- \[.\].* `(.*)`:$',
+                'exitcode': r'^(\s*)- \[.\].* `(.*)`$',
+        }
+
+        command_type = None
+        command_match = None
+
+        for type_c, re_c in command_re.items():
+            match = re.match(re_c, line)
+            if match:
+                command_type = type_c
+                command_match = match
+                break
+
+        if command_type is None:
+            logger.error(f'Failed to match run command line: {line}')
+            return
+
+        command = match[2]
+        output = []
+        insert_offset = self.cursor_offset + 1
+        section_index = -1
+
+        if command_type == 'multi':
+            section_index = self.clean_multi_lines_code()
+
+        try:
+            import subprocess
+
+            proc = subprocess.Popen(command, shell=True, universal_newlines=True,
+                                    bufsize=0,
+                                    stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+
+            while proc.poll() is None:
+                if command_type == 'multi':
+                    while line := proc.stdout.readline():
+                        section_index = self._insert(section_index, line)
+                        output.append(line)
+                        self.writeout()
+                        self._redraw()
+                        self._render()
+                else:
+                    output.extend(proc.stdout.readlines())
+            result = proc.poll()
+        except KeyboardInterrupt:
+            proc.kill()
+            logger.error('execution aborted')
+            return
+        except Exception as e:
+            logger.error(f'Failed to execute `{command}`: {e}')
+            return
+
+        # append the return code to the logs
+        indent = len(match[1])
+        result_line = f'{" " * (indent)} - result code: {int(result)}\n'
+
+        # check if the result was already given:
+        res_match = re.match(f'{" " * (indent)} - result code:.*', self.lines[insert_offset])
+        if res_match:
+            self.lines[insert_offset] = result_line
+            insert_offset += 1
+        else:
+            insert_offset = self._insert(insert_offset, result_line)
+        self.writeout()
+        self._redraw()
+
+        if command_type == 'exit_code':
+            self.mark()
+            # no need to do anything more for exit code
+            return
```

Comment:
```suggestion
        if command_type == 'exit_code':
            # no need to do anything more for exit code
            pass
```
and then have `self.mark()` before the exit of the function, so it reads as:
```python
if command_type == 'exit_code':
   pass
elif ...
elif ...
else:
    return

self.mark()
```